packages/atomic/src/components/atomic-search-box/atomic-search-box.tsx

```diff
@@ -99,6 +118,25 @@ export class AtomicSearchBox {
     this.pendingSuggestionEvents = [];
   }
 
+  public componentDidUpdate() {
+    if (!('redirectTo' in this.searchBoxState)) {
+      return;
+    }
+
+    const {redirectTo, value, analytics} = this.searchBoxState;
+
+    if (redirectTo) {
+      const data = {value, analytics};
+      const storage = new SafeStorage();
+      storage.setItem(
+        StorageItems.STANDALONE_SEARCH_BOX_DATA,
+        JSON.stringify(data)
+      );
+
+      window.location.href = redirectTo;
+    }
```

Comment:
I slightly prefer not having a `return` in a lifecycle method (or a constructor). Perhaps this should be extracted into a separate methods?

E.g.:
```ts
if (this.isStandaloneSearchBox()) {
  const {redirectTo, value, analytics} = this.searchBoxState;
  if (redirectTo) {
    this.redirectTo(redirectTo, value, analytics);
  }
}
```

Reply:
I would agree if we used the lifecycle method for other purposes, but here what would it change?
```
componentDidUpdate() {
 redirect();
}
```